### ISSUE TITLE
Prefer type correct auto import for quick fixes

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#42615]
# Bug Report

From https://github.com/microsoft/vscode/issues/115370

### 🔎 Search Terms

- Quick fix
- auto import

### 🕗 Version & Regression Information

**Tested version**: 4.2.0-dev.20210201

### ⏯ Playground Link

N/A

### 💻 Code
For the project:

```ts
// a1.ts
export const abc = 123
```

```ts
// a2.ts
export const abc = '123'
```

```ts
// index.ts
function foo(x: string) { }

foo(abc); // trigger quick fix here
```

1. Trigger the quick fix on `abc` in `foo(abc)`

### 🙁 Actual behavior

The import quick fixes are returned in file order:

<img width="534" alt="Screen Shot 2021-02-02 at 12 45 53 PM" src="https://user-images.githubusercontent.com/12821956/106660281-a1bb5880-6554-11eb-9e64-20f1c343aefb.png">

Accepting the first suggestion results in invalid code:

<img width="868" alt="Screen Shot 2021-02-02 at 12 46 42 PM" src="https://user-images.githubusercontent.com/12821956/106660334-b5ff5580-6554-11eb-8e4b-cdca3b18fb1b.png">

### 🙂 Expected behavior

In this case, we should sort the import for `a2` before `a1` since it has the correct type

**Comment by DanielRosenwasser** [comment]
I could see us just filtering out, or re-sorting suggestions based on compatibility

**Comment by DanielRosenwasser** [comment]
You definitely don't want to filter since you might be writing part of a bigger expression.

**Comment by mjbvz** [comment]
Suggestions are tricky because the best sort order could change as you type. VS Code supports re-requesting suggestions with every keystroke, but this is probably not something we can do for JS/TS

For quick fixes, I think we should still probably include imports with incorrect types but make sure they come after the better imports. It's possible for example that you would want to use a specific symbol and don't realize it has the wrong type. In that case, it would be nice to still show the suggestion

**Comment by RyanCavanaugh** [comment]
I don't see the value proposition here. We have no idea that you're not about to write a property access that would yield the correct type, so this is a battery sink that makes the order less predictable (which is really bad for muscle memory).

**Comment by mjbvz** [comment]
@RyanCavanaugh Yes, thinking about this more I agree this is probably not feasible for auto import suggestions right now. 

For quick fixes though, the lightbulb generally should not produce invalid code so I think it does make sense to be smarter for those. Quick fixes are also being requested less often, so we don't have the same performance concerns as with suggestions.

In VS Code, we also have an 'auto fix' command that will try to automatically apply the best quick fix to address a problem. It would be great if auto fix knew to use the type correct import by default instead of always defaulting to the first returned import suggestion

**Comment by RyanCavanaugh** [comment]
That distinction makes sense 👍
